client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/SyncModeTransactionTable.java

```diff
@@ -134,15 +134,15 @@ public void beforeCompletion() {
       public void afterCompletion(int status) {
          if (trace) {
             log.tracef("AfterCompletion(xid=%s, status=%s, remote-caches=%s)", xid, transactionStatusToString(status),
-                  preparedCaches);
+                  registeredCaches.keySet());
          }
-         //we only use the preparedCaches. the remaining caches are read-only or they didn't contact the server.
+         TransactionContext<?, ?> ctx = registeredCaches.values().iterator().next();
+         //the server commits everything when the first request arrives.
          try {
             boolean commit = status == Status.STATUS_COMMITTED;
-            for (TransactionContext<?, ?> ctx : preparedCaches) {
-               ctx.complete(xid, commit);
-            }
+            ctx.complete(xid, commit);
          } finally {
+            ctx.forget(xid); //no recovery sync
```

Comment:
Since the commit is a single operation sent to a single server, I'm not sure we need a separate forget operation after a successful commit.

Reply:
in case if the client times-out and the retry kicks in, it finds the transaction in server with committed state.

Reply:
ok. I can change for CR1. this is Synchronization mode anyway...